tests/python/gaia-ui-tests/gaiatest/apps/keyboard/app.py

```diff
@@ -335,15 +327,7 @@ def dismiss(self):
         self.marionette.execute_script("""
 var keyboard = navigator.mozKeyboard || navigator.mozInputMethod;
 keyboard.removeFocus();""")
-        input_window = self.marionette.find_element(*self._input_window_locator)
-
-        # if we have software buttons, keyboar's y will not be 0 but the minus height of the button container.
-        expected_y = int(input_window.size['height']) - System(self.marionette).software_buttons_height
-
-        Wait(self.marionette).until(
-            lambda m: 'inactive' in input_window.get_attribute('class') and
-            not input_window.is_displayed() and
-            (int(input_window.location['y']) == expected_y))
+        Wait(self.marionette).until(lambda m: self.is_keyboard_displayed is False)
```

Comment:
So this would return false when only one of the conditions is true, I guess that would be sufficient to consider the keyboard is not visible from the screen?